### PR TITLE
Add Cloud Support to Master

### PR DESCRIPTION
**Pull request by kenhill**
Replacement for #493 

Disables a few option in Kconfig when PRFLOW is selected and adds a default defconfig for cloud users

### CODE REVIEW

File: hardware/Makefile [hunk 1]
```diff
@@ -286,7 +286,12 @@ hw_project: hw_project_start action_hw
...
-	@if [ `echo "$(USE_PRFLOW)" | tr a-z A-Z` = "TRUE" ]; then \
+ifeq ($(USE_PRFLOW),TRUE)
+	@echo -e "[BUILD IMAGE.........] start `date +"%T %a %b %d %Y"`\n"
+	snap-cloud-build ${SNAP_ROOT}/snap_env.sh
+	@echo -e "[BUILD IMAGE.........] end `date +"%T %a %b %d %Y"`\n"
+else
+  @if [ `echo "$(USE_PRFLOW)" | tr a-z A-Z` = "TRUE" ]; then \
 		echo; echo -e "Error: Makefile target $* not allowed for PR flow!"; exit -1; \
 	fi
 	@echo -e "[BUILD IMAGE.........] start `date +"%T %a %b %d %Y"`\n"
```
Comment (boekholt) [inline hardware/Makefile]: @kenhill These lines are not required anymore.
I did create pull request #498 into this branch containing a suggested change for this.
Please merge this change into this branch or create a different version if you don't like my change.

File: snap_env [hunk 2]
```diff
@@ -94,7 +94,9 @@ while [ -z "$SETUP_DONE" ]; do
   if [ -e "$PSL_DCP" ]; then
     # checking card type for PSL design checkpoint
     PSL_DCP_TYPE=`$snapdir/hardware/snap_check_psl $PSL_DCP`
-    if [ "$PSL_DCP_TYPE" != "$FPGACARD" ]; then
+    if [ "$USE_PRFLOW" == "TRUE" ]; then
+      SETUP_INFO="$SETUP_INFO\n### INFO ### USE_PRFLOW is skipping PSL_DCP check"
```
Comment (boekholt) [inline snap_env]: I think PSL_DCP is required in step `clould_base`

**Comment by kenhill** [comment]
Yes, PSL_DCP is needed for cloud_base. For now, cloud_base is primarily
used internally by the cloud_provider and not the user.

The motivation for this change is to allow users to use snap_config under
the PRFLOW without needing to change PSL_DCP. What do you think about
separating cloud_build from cloud_action/cloud_merge? This would allow us
to remove the PSL_DCP check entirely for PRFLOW and still enforce it with
cloud_base.

**Comment by kenhill** [comment]
@boekholt, I removed psl_dcp from the Vivado project when cloud_user_flow is TRUE. Please pull the branch into master if you approve of the changes.

**Comment by boekholt** [comment]
Yes, we just created release tag v1.2.0 (https://github.com/open-power/snap/tree/v1.2.0)

**Pull request merged**